resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -80,40 +80,39 @@ protected void initialize(Class type, Type genericType, String paramName, Class
       baseType = type;
       baseGenericType = genericType;
 
-      if (type.isArray()) baseType = type.getComponentType();
-      if (List.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = ArrayList.class;
-      }
-      else if (SortedSet.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = TreeSet.class;
-      }
-      else if (Set.class.isAssignableFrom(type))
+      if (type.isArray())
       {
-         isCollection = true;
-         collectionType = HashSet.class;
+    	  baseType = type.getComponentType();
       }
-      if (isCollection)
+     
+      if (!baseType.isPrimitive())
       {
-         if (genericType != null && genericType instanceof ParameterizedType)
+         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
+         if (paramConverter != null) 
          {
-            ParameterizedType zType = (ParameterizedType) genericType;
-            baseType = Types.getRawType(zType.getActualTypeArguments()[0]);
-            baseGenericType = zType.getActualTypeArguments()[0];

```

Comment:
Hi Ron,

Yes this is it and that's the kind of tests done in `org.jboss.resteasy.test.resource.param.ParamConverterTest.testIt()` unit tests too.

I'm not saying that another synthax is really mandatory (for sure it's not). 
Just that it will be nice to provide support for it since it does not require a lot of code modification to handle it so that users can choose wheiter doing "...?q=a&q=b" or "...?q=a,b".

 The "...?q=a&q=b" is for sure the cleanest syntax and will be the one supported by default (just like now). But if users want to take advantage of the second one  by fulfilling JaxRS spec conditions listed previously why not ? 

More, according to the specs users are free to regsiter a `ParamConverter` to handle `List<Date>` for example. So they can use any syntax since they are able to parse it into `List<Date>`